### DISCUSSION TITLE
Any plans to implement LZW or JPEG (old or new) compression?

### DISCUSSION THREAD

**Discussion by Echoloc8** [ngageoint/tiff-java#21] [Ideas]
I'm using this library on Android, and while I can see results with AdobeDeflate compression, the other tools I will need to use this functionality with cannot. When trying JPEG and LZW compression I got a nice clean "not compatible" or "not implemented" message.

I'm sure the possibility is there for me to implement them myself, but I was hoping for this library to save some time. :-)

Any possibility of these compression types being added?

**Comment by bosborn** [comment]
Nothing planned in the near future simply based on our own needs and schedules, but always a possibility.
We are open to pull requests if you want to implement for this library.

[LZW Encoding](https://github.com/ngageoint/tiff-java/blob/4a5f3257782eab5670154b3bf7213d1d345ee11e/src/main/java/mil/nga/tiff/compression/LZWCompression.java#L293)
[JPEG Encoder Starting Point](https://github.com/ngageoint/tiff-java/blob/4a5f3257782eab5670154b3bf7213d1d345ee11e/src/main/java/mil/nga/tiff/TiffWriter.java#L459)
[JPEG Decoder Starting Point](https://github.com/ngageoint/tiff-java/blob/4a5f3257782eab5670154b3bf7213d1d345ee11e/src/main/java/mil/nga/tiff/FileDirectory.java#L150)

**Comment by bosborn** [comment]
_Converted from an Issue to a Discussion_
